eos_data_distribution/avahi_monitor.py

```diff
@@ -184,13 +187,22 @@ def service_removed_cb(self, sda, interface, protocol, name, type, domain, flags
             self.process_route_changes()
 
     def add_nexthop(self, faceUri):
+        if len(self._peers) > MAX_PEERS:
+            logger.debug(
+                "refusing to add %s as we'd go over %d peers", faceUri, MAX_PEERS)
+            return
+        self._peers.add(faceUri)
         return check_call(["nfdc", "add-nexthop",
                            "-c", str(defaults.RouteCost.LOCAL_NETWORK),
                            str(SUBSCRIPTIONS_SOMA), faceUri])
```

Comment:
There is no binding between the success of the `nfdc add-nexthop` invocation and the `self._peers.add()` call here: the peer will always be added to `self._peers`, but then the `add-nexthop` invocation could fail, leaving the state out of sync.